packages/open-next/src/adapters/server-adapter.ts

```diff
@@ -0,0 +1,271 @@
+  debug("ServerResponse data", { statusCode, headers, isBase64Encoded, body });
+
+  // WORKAROUND: `NextServer` does not set cache response headers for HTML pages — https://github.com/serverless-stack/open-next#workaround-nextserver-does-not-set-cache-response-headers-for-html-pages
+  if (htmlPages.includes(parser.rawPath) && headers["cache-control"]) {
+    headers["cache-control"] =
+      "public, max-age=0, s-maxage=31536000, must-revalidate";
+  }
+
+  return isCloudFrontEvent
+    ? formatCloudFrontResponse({ statusCode, headers, isBase64Encoded, body })
+    : formatApiv2Response({ statusCode, headers, isBase64Encoded, body });
+}
+
+//////////////////////
+// Helper functions //
+//////////////////////
+
+function setNextjsServerWorkingDirectory() {
+  // WORKAROUND: Set `NextServer` working directory (AWS specific) — https://github.com/serverless-stack/open-next#workaround-set-nextserver-working-directory-aws-specific
+  process.chdir(__dirname);
+}
+
+function loadHtmlPages() {
+  const filePath = path.join(nextDir, "server", "pages-manifest.json");
+  const json = fs.readFileSync(filePath, "utf-8");
+  return Object.entries(JSON.parse(json))
+    .filter(([_, value]) => (value as string).endsWith(".html"))
+    .map(([key]) => key);
+}
+
+async function processRequest(req: IncomingMessage, res: ServerResponse) {
+  // @ts-ignore
+  // Next.js doesn't parse body if the property exists
+  // https://github.com/dougmoscrop/serverless-http/issues/227
+  delete req.body;
+
+  try {
+    await requestHandler(req, res);
+  } catch (e: any) {
+    console.error("NextJS request failed.", e);
+
+    res.setHeader("Content-Type", "application/json");
+    res.end(
+      JSON.stringify(
+        {
+          message: "Server failed to respond.",
+          details: e,
+        },
+        null,
+        2
+      )
+    );
+  }
+}
+
+function formatApiv2Response({
+  statusCode,
+  headers: rawHeaders,
+  body,
+  isBase64Encoded,
+}: {
+  statusCode: number;
+  headers: Record<string, string | string[]>;
+  body: string;
+  isBase64Encoded: boolean;
+}) {
+  const headers: Record<string, string> = {};
+  Object.entries(rawHeaders)
+    .filter(([key]) => key.toLowerCase() !== "set-cookie")
+    .forEach(([key, value]) => {
+      if (value === null) {
+        headers[key] = "";
+        return;
+      }
+      headers[key] = Array.isArray(value) ? value.join(", ") : value.toString();
+    });
+  const response: APIGatewayProxyResultV2 = {
+    statusCode,
+    headers,
+    cookies: rawHeaders["set-cookie"] as string[] | undefined,
+    body,
+    isBase64Encoded,
+  };
+  debug(response);
+  return response;
+}
+
+function formatCloudFrontResponse({
+  statusCode,
+  headers: rawHeaders,
+  body,
+  isBase64Encoded,
+}: {
+  statusCode: number;
+  headers: Record<string, string | string[]>;
+  body: string;
+  isBase64Encoded: boolean;
+}) {
+  const headers: CloudFrontHeaders = {};
```

Comment:
```suggestion
  const headers: CloudFrontHeaders = {};
  delete rawHeaders["content-length"]; // CF throws 502 if "content-length" is in headers
```

Reply:
Do you have a source for the comment? The CF docs explicitly recommend adding a content-length header:

https://docs.aws.amazon.com/AmazonCloudFront/latest/DeveloperGuide/RequestAndResponseBehaviorCustomOrigin.html

> We recommend that you configure your HTTP server to add a Content-Length header to prevent CloudFront from caching partial objects.

Reply:
I wish I did. Maybe the correct approach is the remove the `content-length` on specific request types? All my api requests (including next-auth) would throw a 502 error. I didn't research further on why that was happening though.

Reply:
I read it here — https://docs.aws.amazon.com/AmazonCloudFront/latest/DeveloperGuide/edge-functions-restrictions.html#function-restrictions-read-only-headers-origin-request